### PR TITLE
Fix/635 deployment of the library to npm

### LINKED ISSUE #635: Fix deployment of the library to npm
- bring back npm Bazel rules 
- fix to enable deploying to npm while keeping the lib in the monorepo

### PR DESCRIPTION
**Pull request by AudreyKj**
closes #635

### CODE REVIEW

File: lib/typescript/httpclient/BUILD [hunk 1]
```diff
@@ -13,6 +15,51 @@ ts_web_library(
...
+web_library(
+    name = "dist",
+    app_lib = ":httpclient",
+    entry = "typescript/httpclient/index.ts",
+    externals = {
+        "@types/node": "@types/node",
+        "camelcase-keys": "camelcase-keys",
+    },
+    output = {
+        "library": "@airyhq/httpClient",
```
Comment (chrismatix) [inline lib/typescript/httpclient/BUILD:27]: Npm modules are usually in `kebab-case`. I think we should go with that convention.

File: tools/build/npm/rules.bzl [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        "version_file": attr.label(
+            allow_single_file = True,
+            doc = """
+            File containing version string.
+            Alternatively, pass --define version=VERSION to Bazel invocation.
+            Not specifying version at all defaults to '0.0.0'
+            """,
+        ),
+        "_assemble_script": attr.label(
+            default = "//npm:assemble",
```
Comment (chrismatix) [inline tools/build/npm/rules.bzl:69]: This package label needs to be fixed to `//tools/build/npm:assemble`

File: tools/build/npm/rules.bzl [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            mandatory = True,
+            doc = "Snapshot repository to deploy npm artifact to",
+        ),
+        "release": attr.string(
+            mandatory = True,
+            doc = "Release repository to deploy npm artifact to",
+        ),
+        "_deployment_script_template": attr.label(
+            allow_single_file = True,
+            default = "//npm/templates:deploy.py",
```
Comment (chrismatix) [inline tools/build/npm/rules.bzl:129]: Same here

File: tools/build/npm/templates/deploy.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+npm_repositories = {
+    "snapshot": "{snapshot}",
+    "release": "{release}",
+}
+npm_registry = npm_repositories[repo_type]
+
+npm_username, npm_password, npm_email = (
+    os.getenv('DEPLOY_NPM_USERNAME'),
+    os.getenv('DEPLOY_NPM_PASSWORD'),
+    os.getenv('DEPLOY_NPM_EMAIL'),
```
Comment (chrismatix) [inline tools/build/npm/templates/deploy.py:48]: Note: Once you add the CI deployment you will have to provide these three variables as secrets.

**Review by chrismatix — COMMENTED** [review]
Looks good sofar. The build does not work yet so I left some comments

File: lib/typescript/httpclient/index.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
-export * from './endpoints';
+export * from './scr/endpoints';
```
Comment (chrismatix) [inline lib/typescript/httpclient/index.ts:1]: Small typo `src`

File: lib/typescript/httpclient/package.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "@airyhq/http-client",
+  "version": "0.1.0",
+  "description": "Airy http client library",
+  "main": "dist/index.js",
+  "browser": {
+    "./dist/index.js": "./dist/index.browser.js",
+    "./dist/index.esm.js": "./dist/index.browser.esm.js"
+  },
```
Comment (chrismatix) [inline lib/typescript/httpclient/package.json:9]: Nice 👍

File: lib/typescript/httpclient/yarn.lock [hunk 7]
```diff
@@ -0,0 +1,32 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (chrismatix) [inline lib/typescript/httpclient/yarn.lock:1]: I am wondering if it makes sense to commit this file since we don't use it for the bazel builds

**Review by chrismatix — APPROVED** [review]
LGTM with a minor nitpick.

File: lib/typescript/httpclient/yarn.lock (hunk 7 above)
Comment (AudreyKj) [reply lib/typescript/httpclient/yarn.lock:1]: makes sense! will take it out!

**Review by chrismatix — APPROVED** [review]

**Pull request merged**